### PR TITLE
fix: Fixed bugs for analyzing packages and api data creation.

### PR DESCRIPTION
**Pull request by Masara**
### Summary of Changes
* Fixed a bug in `_get_api.py` which prevented analyzing packages with different path lengths.
* Fixed a bug in `_ast_visitor.py` for the api data creation, in which the id of modules did not correctly represent their path.
* Adjusted test snapshots

### CODE REVIEW

File: tests/safeds_stubgen/__snapshots__/test_main.ambr [hunk 1]
```diff
@@ -602,6 +602,21 @@
           'name': 'int',
         }),
       }),
+      dict({
+        'docstring': dict({
+          'default_value': '',
+          'description': '',
+          'type': '',
+        }),
+        'id': 'test_package/test_package/test_package/test_package/test_package',
```
Comment (lars-reimann) [inline tests/safeds_stubgen/__snapshots__/test_main.ambr:611]: Is this intended?

File: tests/data/test_package/test_package/test_package.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class TestPackage:
```
Comment (lars-reimann) [inline tests/data/test_package/test_package/test_package.py:1]: Is it intended that all parent folders and API elements are some variation of `TestPackage`? I find it rather hard to validate the snapshots with those names.

File: src/safeds_stubgen/api_analyzer/_ast_visitor.py [hunk 3]
```diff
@@ -95,13 +95,22 @@ def enter_moduledef(self, node: MypyFile) -> None:
...
 
+        # Create module id to get the full path
+        module_id_parts = node.fullname.split(self.api.package)[1:]
+        module_id = self.api.package.join(module_id_parts)
+        if module_id.startswith("."):
+            module_id = module_id[1:]
+        module_id = module_id.replace(".", "/")
+        formatted_module_id = f"{self.api.package}/{module_id}"
+        id_ = self.__get_id(formatted_module_id)
+
```
Comment (lars-reimann) [inline src/safeds_stubgen/api_analyzer/_ast_visitor.py:106]: Please extract this into its own function.

File: src/safeds_stubgen/api_analyzer/_ast_visitor.py [hunk 4]
```diff
@@ -95,13 +95,22 @@ def enter_moduledef(self, node: MypyFile) -> None:
             elif isinstance(definition, ExpressionStmt) and isinstance(definition.expr, StrExpr):
                 docstring = definition.expr.value
 
+        # Create module id to get the full path
+        module_id_parts = node.fullname.split(self.api.package)[1:]
+        module_id = self.api.package.join(module_id_parts)
```
Comment (lars-reimann) [inline src/safeds_stubgen/api_analyzer/_ast_visitor.py:100]: This seems wrong: `"a".join(["b", "c"])` produces `"abac"`. The `split` in the line above also looks fishy.

File: src/safeds_stubgen/api_analyzer/_ast_visitor.py [hunk 5]
```diff
@@ -95,13 +95,22 @@ def enter_moduledef(self, node: MypyFile) -> None:
...
+        if module_id.startswith("."):
+            module_id = module_id[1:]
+        module_id = module_id.replace(".", "/")
+        formatted_module_id = f"{self.api.package}/{module_id}"
+        id_ = self.__get_id(formatted_module_id)
```
Comment (lars-reimann) [inline src/safeds_stubgen/api_analyzer/_ast_visitor.py:105]: What does `__get_id` do? Please add a docstring. In any case, passing an ID to a `__get_id` function and getting an ID back seems weird.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds_stubgen/api_analyzer/_ast_visitor.py (hunk 5 above)
Comment (lars-reimann) [reply src/safeds_stubgen/api_analyzer/_ast_visitor.py:105]: Also: A single underscore suffices to mark a method as internal, i.e. `_get_id`. A double underscore prefix does some magic on the names of class methods.

Comment (Masara) [reply src/safeds_stubgen/api_analyzer/_ast_visitor.py]: Added a docstrings, renamed the ```__get_id``` function and adjusted the function.

File: src/safeds_stubgen/api_analyzer/_ast_visitor.py (hunk 4 above)
Comment (Masara) [reply src/safeds_stubgen/api_analyzer/_ast_visitor.py]: I don't quite understand the problem here, could you elaborate it a bit further? I added code comments if the code was unclear.

File: tests/safeds_stubgen/__snapshots__/test_main.ambr (hunk 1 above)
Comment (Masara) [reply tests/safeds_stubgen/__snapshots__/test_main.ambr:611]: Removed these snapshot tests and added unit tests instead.

File: tests/data/test_package/test_package/test_package.py (hunk 2 above)
Comment (Masara) [reply tests/data/test_package/test_package/test_package.py:1]: Removed these snapshot tests and added unit tests instead.

**Comment by Masara** [comment]
@lars-reimann Everything should be okay now, I don't know how to resolve the changes requests though. Could you review and merge once again?

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**